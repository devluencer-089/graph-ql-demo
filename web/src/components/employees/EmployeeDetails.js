import React from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import { withStyles } from 'material-ui/styles';

export const query = gql`
  {
    query
    goes
    here
  }
`;

const styles = theme => ({
  root: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginTop: theme.spacing.unit * 3
  })
});

function EmployeeDetails(props) {
  //return <Query query={}/>

  return <div />;
}

export default withStyles(styles)(EmployeeDetails);
