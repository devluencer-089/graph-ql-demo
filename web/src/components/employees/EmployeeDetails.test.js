import React from 'react';
import { createShallow } from 'material-ui/test-utils';
import EmployeeDetails, { query } from './EmployeeDetails';
import { MockedProvider } from 'react-apollo/test-utils';

describe('EmployeeDetails Component', () => {
  let shallow = createShallow({ dive: true });
  let data = {
    getEmployee: {
      id: '3',
      level: 'SENIOR_DEVELOPER',
      firstname: 'Si',
      lastname: 'Tran',
      projects: [{ name: 'TeamBank' }, { name: 'Paydirekt' }]
    }
  };
  let match = {
    params: {
      employeeId: 3
    }
  };

  it('should match snapshot', () => {
    let eut = shallow(
      <MockedProvider>
        <EmployeeDetails data={data} classes={{}} match={match} />
      </MockedProvider>
    );

    expect(eut).toMatchSnapshot();
  });

  describe('GraphQL Query', () => {
    it('should match snapshot', () => {
      expect(query).toMatchSnapshot();
    });
  });
});
