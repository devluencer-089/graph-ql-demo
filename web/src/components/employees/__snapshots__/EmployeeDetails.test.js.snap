// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeDetails Component GraphQL Query should match snapshot 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "EmployeeDetailQuery",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "employeeId",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "getEmployee",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "level",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "firstname",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "lastname",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "phoneNumber",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "projects",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "ID",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "employeeId",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 201,
    "start": 0,
  },
}
`;

exports[`EmployeeDetails Component should match snapshot 1`] = `
<Apollo(WithStyles(EmployeeDetails))
  classes={Object {}}
  data={
    Object {
      "getEmployee": Object {
        "firstname": "Si",
        "id": "3",
        "lastname": "Tran",
        "level": "SENIOR_DEVELOPER",
        "projects": Array [
          Object {
            "name": "TeamBank",
          },
          Object {
            "name": "Paydirekt",
          },
        ],
      },
    }
  }
  match={
    Object {
      "params": Object {
        "employeeId": 3,
      },
    }
  }
/>
`;
